include/arch/arm/irq.h

```diff
@@ -136,6 +137,49 @@ static inline void arch_isr_direct_footer(int maybe_swap)
 	} \
 	static inline int name##_body(void)
 
+#if defined(CONFIG_DYNAMIC_DIRECT_INTERRUPTS)
+
+extern void z_arm_irq_direct_dynamic_dispatch_reschedule(void);
+extern void z_arm_irq_direct_dynamic_dispatch_no_reschedule(void);
+
+/**
+ * @brief Macro to register an ISR Dispatcher (with/without re-scheduling
```

Comment:
Here is my take on this:

```

/**
 * @brief Macro to register an ISR Dispatcher (with or without re-scheduling
 * request) for dynamic direct interrupts.
 *
 * This macro registers the ISR dispatcher function for dynamic direct
 * interrupts, allowing the use of dynamic direct ISRs in the kernel.
 * The dispatcher function is invoked when the direct
 * interrupt occurs and then triggers the (soft) Interrupt Service Routine
 * (ISR) that is registered dynamically (i.e. at run-time) into the IRQ table
 * stored in SRAM. The ISR must be connected with irq_connect_dynamic() and
 * enabled via irq_enable() before the dynamic direct interrupt can be
 * serviced. This ISR dispatcher may be configured by the user to trigger
 * thread re-secheduling upon return.
 *
 * These ISRs are designed for performance-critical interrupt handling and do
 * not go through all of the common interrupt handling code. 
 *
 * Dynamic 'direct' interrupts are regular Zephyr interrupts when it comes to their
 * declaration: their signature must match void isr(void* parameter), since, unlike
 * regular direct interrupts, they are not placed directly into the hardware vector
 * table but instead use the software ISR table.
 * The major difference with regular Zephyr interrupts are the following:
 * - Just like direct interrupts, the call into the OS 
 *   to exit power management idle state is optional. Normal interrupts always do this 
 *   before the ISR is run, but when it runs is controlled by the placement of 
 *   a ISR_DIRECT_PM() macro, or omitted entirely.
 * - Similar to direct interrupts, scheduling decisions are optional but must be 
 *   made at build time. They are controlled by @param resch to this macro. 
 *
 *
 * @param irq_p IRQ line number.
 * @param priority_p Interrupt priority.
 * @param flags_p Architecture-specific IRQ configuration flags.
 * @param resch Set flag to 'reschedule' to request thread
 *              re-scheduling upon ISR function. Set flag
 *              'no_reschedule' to skip thread re-scheduling
 *
 * Note: the function is an ARM Cortex-M only API.
 *
 * @return Interrupt vector assigned to this interrupt.
 */
```